lib/jobs/upsert-node-job.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+module.exports = UpsertNodeAttributeFactory;
+
+var di = require('di');
+
+di.annotate(UpsertNodeAttributeFactory, new di.Provide('Job.Upsert.Node.Attribute'));
+di.annotate(UpsertNodeAttributeFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Services.Waterline',
+    'Logger'
+    )
+);
+
+function UpsertNodeAttributeFactory(BaseJob, util, waterline, Logger) {
+    var logger = Logger.initialize(UpsertNodeAttributeFactory);
+
+    function UpsertNodeAttribute(options, context, taskId) {
+        UpsertNodeAttribute.super_.call(this, logger, options, context, taskId);
+        this.nodeId = this.options.target;
+        this.data = this.options.data;
+    }
+    util.inherits(UpsertNodeAttribute, BaseJob);
+    
+    UpsertNodeAttribute.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.updateByIdentifier(self.nodeId, self.data) 
```

Comment:
Actually I tested on my own development environment, it can do both update and insert. 

The test code I used to test the feature is as follows.
[update_node_workflow_tasks.zip](https://github.com/RackHD/on-tasks/files/370247/update_node_workflow_tasks.zip)